test/sideinput-sink-e2e/sideinput-withsink_test.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sideinput_e2e
+
+import (
+	"context"
+	. "github.com/numaproj/numaflow/test/fixtures"
+	"github.com/stretchr/testify/suite"
+	"strconv"
+	"testing"
+	"time"
+)
+
+type SideInputUDSSuite struct {
+	E2ESuite
+}
+
+func (s *SideInputUDSSuite) TestUserDefinedSinkWithSideInput() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	w := s.Given().Pipeline("@testdata/sideinput_sink.yaml").When().CreatePipelineAndWait()
+	defer w.DeletePipelineAndWait()
+	w.Expect().VertexPodsRunning()
+	pipelineName := "user-defined-sink-test"
+	done := make(chan struct{})
+	go func() {
```

Comment:
Do we need to run this go routine? It looks we are not verifying messages sent from it.

Reply:
I am using it to trigger the flow ,without it the keys are not getting created  in the redis container @KeranYang

Reply:
Could you switch to [generator src](https://numaflow.numaproj.io/user-guide/sources/generator/) to trigger?